src/Frame.cpp

```diff
@@ -154,18 +154,17 @@ Frame_ERROR Frame_ERROR::connectionError(const std::string& message) {
   return Frame_ERROR(0, ErrorCode::CONNECTION_ERROR, Payload(message));
 }
 
-Frame_ERROR Frame_ERROR::invalid(
-    StreamId streamId,
-    const std::string& message) {
+Frame_ERROR Frame_ERROR::error(StreamId streamId, Payload&& payload) {
   DCHECK(streamId) << "streamId MUST be non-0";
-  return Frame_ERROR(streamId, ErrorCode::INVALID, Payload(message));
+  return Frame_ERROR(streamId, ErrorCode::INVALID, std::move(payload));
```

Comment:
INVALID is the general error code that something was wrong.
For application errors we are using APPLICATION_ERROR